dags/common/immersion_facilitee/helpers.py

```diff
@@ -0,0 +1,136 @@
+import datetime
+import logging
+import re
+import time
+from typing import Dict, List
+
+import httpx
+import pandas as pd
+from airflow.models import Variable
+from dateutil.relativedelta import relativedelta
+from sqlalchemy.dialects.postgresql import insert as pg_insert
+from sqlalchemy.orm import Session
+
+from dags.common import db
+from dags.common.anonymize__sensible_data import hash_content, normalize_sensible_data
+from dags.common.immersion_facilitee.models import Conventions
+
+
+logger = logging.getLogger(__name__)
+
+
+def api_client() -> httpx.Client:
+    """
+    Crée un client HTTP pour l'API Immersion Facilitee
+    :return: un client HTTP configuré
+    """
+    return httpx.Client(
+        base_url=Variable.get("API_IMMERSION_FACILITEE_BASE_URL"),
+        headers={"authorization": "{}".format(Variable.get("API_IMMERSION_FACILITEE_TOKEN"))},
+        timeout=httpx.Timeout(timeout=5, read=30),
+    )
+
+
+def get_all_items(path: str) -> List[Dict]:
+    """
+    Récupère toutes les conventions des dernières 5 ans
+    :param path: chemin de l'API
+    :return: liste de dictionnaires contenant les conventions
+    """
+
+    # Aujourd'hui on boucle sur les jours car jusqu'à maintenant pour les SIAE il n'y a pas 100
+    # ou plus conventions. L'API IF évoluera probablement permettant la pagination et une date de mise à jour
+
+    start_date_less_or_equal = datetime.date.today()
+    start_date_greater = start_date_less_or_equal - relativedelta(years=5)
+
+    client = api_client()
+
+    date_temp = start_date_greater
+
+    data = []
+    while date_temp < start_date_less_or_equal:
+        str_date_g = date_temp.strftime("%Y-%m-%d")
+        str_date_l = (date_temp + relativedelta(days=1)).strftime("%Y-%m-%d")
```

Comment:
En Python on utilise assez peu `while` ou dans des cas assez précis (bas niveau réseau, boucle infinie, boucle avec plusieurs cas de sortie, etc) car on préfère raisonner en itérable et donc utiliser `for`, ce qui permet de ne pas avoir à se préoccuper d'une variable temporaire ou de la condition de sortie :).

Pour le parcours de date tu as plusieurs alternatives :
1. Utiliser une |rrule](https://dateutil.readthedocs.io/en/stable/rrule.html#module-dateutil.rrule)
2. Utiliser `range()` et `timedelta`/`relativedelta` afin de générer les dates consécutivement